### ISSUE TITLE
OBS Studio Game Capture crash

### ISSUE THREAD

**Issue by undeniablebis** [OpenTTD/OpenTTD#8942]
[crash.log](https://github.com/OpenTTD/OpenTTD/files/6254077/crash.log)
![crash](https://user-images.githubusercontent.com/75613685/113501480-e9ac2b00-9557-11eb-9f45-a02845586f30.png)

**Comment by nielsmh** [comment]
Did this happen as soon as you started capture with OBS, or only after using it for a while?
Are you running windowed or full screen?

(Please fill out the template instead of deleting all the questions. The question are there because they are important.)

**Comment by undeniablebis** [comment]
I’m sorry for deleting the questions. They crash as soon as I started to
capture it. It only happens when I use game capture. When i use window
capture, it shows all the programs that I’m running but it’s freezing. The
game is ok tho. I play it on window mode.

**Comment by nielsmh** [comment]
I can't reproduce this.

Which OBS version?
Is OBS the 32 bit or 64 bit version?
Which graphics card?
What graphics driver version?
Is OpenTTD the 32 bit or 64 bit version?

**Comment by undeniablebis** [comment]
I'm using OBS 64 bit v26.1.1
My graphics card is GeForce 720m v388.73
OpenTTD is 64bit

Sorry for the late reply, I was cooking dinner.

**Comment by TrueBrain** [comment]
There should also be a `crash.dmp` file; could you zip that and attach it to your post?

**Comment by nielsmh** [comment]
Really, the crash.log file already indicates that the crash happens deep inside the NVidia driver, so my first suggestion would be to try to update that, if possible.

**Comment by undeniablebis** [comment]
[crash.zip](https://github.com/OpenTTD/OpenTTD/files/6254378/crash.zip)

**Comment by TrueBrain** [comment]
@undeniablebis : a question and two possible solutions for you:

First, the question:
- Does this also happen with other games, or only with OpenTTD?

Next, two things come to mind that might resolve the issue for you:
- The driver you have loaded is from 2017. I can imagine there have been fixes over the years that address similar problems. If possible, it might be good to upgrade your NVidia drivers. There should be an update available for your GPU, although it is from 2018, so that might still be a problem :)
- You can disable "hardware acceleration" in the game (via Game Options). This means the "Game Capture" will no longer work (as OpenTTD will no longer use OpenGL), but the Window Capture should work fine, without the freezing part. It is a compromise.

Additional, something that I do not think has any effect, but nevertheless might be worth a try:
You can start the game with `-vwin32-opengl:no_threads`. Most commonly, one can do this via Command Prompt or Powershell. Navigate to the folder `cd 'C:\Program Files\OpenTTD\'` and then execute `openttd.exe -vwin32-opengl:no_threads`. I do not think this is going to make any difference, but it might be worth a try.

**Comment by mattkimber** [comment]
I tried reproducing this on my 870m laptop, as I've had similar issues relating to Optimus on the mobile chipsets. Unfortunately a "cannot reproduce" with that config, using both:

- OBS and OpenTTD on Nvidia GPU (I can capture fine using the stated version of OBS)
- OBS on Nvidia, OpenTTD on Intel integrated graphics (my capture is a black screen, but it doesn't crash)

I can't seem to get OBS to start on integrated no matter what settings I try to configure in the NVidia driver, it gets forced to the NVidia GPU.

One thing to note is I have a much newer driver, 425.31, so updating may help. (I've noticed fewer Optimus-related crashes generally compared to when I first got this laptop)

**Comment by Andrew350** [comment]
> I’m sorry for deleting the questions. They crash as soon as I started to
> capture it. It only happens when I use game capture. When i use window
> capture, it shows all the programs that I’m running but it’s freezing. The
> game is ok tho. I play it on window mode.

Maybe try closing other programs and/or doing a system restart?

I had this exact issue a couple days ago when trying to record the RC1 build, unfortunately in my case when OpenTTD crashed it didn't produce a crash report (just locked up the window). I never tried game capture, only window capture, but this was also causing OBS to lock up very frequently the same way you describe (with or without OpenGL active). Another game I tried at the time did not cause OBS to crash (or crash itself).

My specs:
Windows 10
GTX 1660Ti (laptop), driver version 461.40
OBS v25.0.8
Everything 64-bit

(Un)fortunately, everything now seems to run perfectly fine. I just tried various configs of OpenTTD and OBS for almost an hour trying to reproduce it with no luck, whereas before OBS would crash as soon as I hit a settings button, and OpenTTD would (very rarely) lock up after only a minute or two of trying to record.

The only thing I can think of that has changed, is that the next day I noticed I had GIMP running in the background which had apparently developed a memory leak or something, because my whole system began stuttering every 10 seconds or so. Once I killed GIMP, my system runs fine, and since then I have not witnessed the crashing/freezing behavior between OBS and OpenTTD.

It may just be a huge coincidence, but until I can reproduce the behavior it appears, at least in my case, that another program may have somehow been causing the issue, so there's a small chance this may be your issue too 🙂
